packages/bundler/src/BundlerCollectorTracer.ts

```diff
@@ -108,15 +129,28 @@ export function bundlerCollectorTracer (): BundlerCollectorTracer {
     step (log: LogStep, db: LogDb): any {
       const opcode = log.op.toString()
       // this.debug.push(this.lastOp + '-' + opcode + '-' + log.getDepth())
-      if (opcode === 'NUMBER') this.numberCounter++
-      if (this.numberLevels[this.numberCounter] == null) {
-        this.currentLevel = this.numberLevels[this.numberCounter] = {
-          access: {},
-          opcodes: {}
-        }
+
+      if (opcode === 'REVERT' || opcode === 'RETURN') {
+        const ofs = parseInt(log.stack.peek(0).toString())
+        const len = parseInt(log.stack.peek(1).toString())
+        const data = toHex(log.memory.slice(ofs, ofs + len)).slice(0, 500)
```

Comment:
Just saying - are you going to parse these 500 characters back to buffer? Maybe append '...' to indicate data is truncated?

Reply:
hmm. later I decode the buffer, so appending "..." will break it.
i truncate it so that I don't inadvertently return a huge buffer created by some contract.